shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/SecretRestartHandler.java

```diff
@@ -0,0 +1,50 @@
+package com.redhat.service.smartevents.shard.operator;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.runtime.Quarkus;
+import io.quarkus.runtime.Startup;
+import io.quarkus.scheduler.Scheduled;
+
+@Startup
+@ApplicationScoped
+public class SecretRestartHandler {
```

Comment:
For other reviewers:

When we install the Shard operator via the AddOn flow into an OSD cluster, the parameters a user specifies to configure the AddOn are automatically provisioned into a `Secret` called `addon-smart-events-operator-parameters`. The challenge is that there is no way for us to restart our AddOn installed operator in response to changes in configuration without manual intervention.

This simple loop watches the external `Secret` we configure and simply restarts the Shard should the content change. This will also help us resolve https://issues.redhat.com/browse/MGDOBR-229